packages/flutter_tools/lib/src/build_system/targets/darwin.dart

```diff
@@ -152,3 +154,96 @@ abstract class CheckDevDependencies extends Target {
     globals.stdio.stderrWrite('error: $message');
   }
 }
+
+abstract class UnpackDarwin extends Target {
+  const UnpackDarwin();
+
+  Future<void> copyFramework(
```

Comment:
I assume this should only be called by subtypes? If so, consider adding the `@protected` annotation: https://api.flutter.dev/flutter/meta/protected-constant.html

```suggestion
  @protected
  Future<void> copyFramework(
```

Same suggestion to `thinFramework` as well